### PR TITLE
Add github action tests for native workers (presto_cpp)

### PR DESCRIPTION
**Pull request by mshang816**
Add github action tests for native workers (presto_cpp)
Here is a summary of the changes:
- Pull latest changes from presto_cpp and velox
- Add a new github action job in .github/workflows/test-native.yml that runs unit tests and e2e tests for native workers. Since cache has been enabled for the workflow, the entire test will finish in about 20 mins when the cache is warm.
- Add presto-native-execution module to the root pom.xml. This way we can run the e2e from the presto project root as presto_cpp (presto-native-execution) is now part of presto. And its version should get updated along with other presto components as the release is being rolled out, so we are free from maintaining the version as we did in presto_cpp repository.
- After the PR is merged, we can get signals to see if the native worker is broken by a PR (either the change is on Java or C++ side)

Currently, presto-native-execution contains both C++ worker and Java e2e test code. It's more convenient for us to synchronize the code and tests in one pass to fbcode. We may consider extracting the java test code to its standalone component later.

Test plan - (Please fill in how you tested your changes)
Check if the newly added job `test native / build-and-test` runs fine from the "Checks" tab of this PR.

== NO RELEASE NOTE ==

### CODE REVIEW

**Review by spershin — APPROVED** [review]
I don't understand much in the code, but it works and looks good to me. :)

**Comment by mbasmanova** [comment]
@mshang816 Michael, there are a few typos in the PR description. Would you fix these and edit the description to either add release notes or use == NO RELEASE NOTE ==?

**Comment by mbasmanova** [comment]
I see `test native / build-and-test (pull_request) Successful in 34m` job. Is this the job being added here?

**Comment by mshang816** [comment]
> @mshang816 Michael, there are a few typos in the PR description. Would you fix these and edit the description to either add release notes or use == NO RELEASE NOTE ==?

I've updated the descriptions to include "== NO RELEASE NOTE ==" as there is no client-facing changes.

**Comment by mbasmanova** [comment]
@mshang816 Michael, would you also fix the typos and remove the following boiler plate from the description?

```
Test plan - (Please fill in how you tested your changes)

Please make sure your submission complies with our [Development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [Formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), and [Commit Message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests) guidelines. Don't forget to follow our [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution) for any code copied from other projects.

Fill in the release notes towards the bottom of the PR description.
See [Release Notes Guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) for details.
```

**Comment by mshang816** [comment]
> I see `test native / build-and-test (pull_request) Successful in 34m` job. Is this the job being added here?

That is correct.

File: presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestHiveAggregationQueries.java [hunk 1]
```diff
@@ -50,7 +50,7 @@ public void testAggregations()
         // non-deterministic query
         assertQuerySucceeds("SELECT orderkey, arbitrary(comment) FROM lineitem GROUP BY 1");
 
-        assertQuery("SELECT orderkey, array_agg(linenumber) FROM lineitem GROUP BY 1");
+        // assertQuery("SELECT orderkey, array_agg(linenumber) FROM lineitem GROUP BY 1");
```
[the remark this answers is not in the document]
Comment (mshang816) [reply presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestHiveAggregationQueries.java:53]: @mbasmanova somehow this query failed during the e2e test so I commented this one out for the time being as it's not the main concern of this change. The test code will get updated to the lastest from presto_cpp when I pull in the code next time. Note that this is still something in progress and not officially made public yet.

**Comment by mshang816** [comment]
> @mshang816 Michael, would you also fix the typos and remove the following boiler plate from the description?
> 
> ```
> Test plan - (Please fill in how you tested your changes)
> 
> Please make sure your submission complies with our [Development](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development), [Formatting](https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#formatting), and [Commit Message](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#commit-formatting-and-pull-requests) guidelines. Don't forget to follow our [attribution guidelines](https://github.com/prestodb/presto/wiki/Review-and-Commit-guidelines#attribution) for any code copied from other projects.
> 
> Fill in the release notes towards the bottom of the PR description.
> See [Release Notes Guidelines](https://github.com/prestodb/presto/wiki/Release-Notes-Guidelines) for details.
> ```

done

File: presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestHiveAggregationQueries.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (mbasmanova) [reply presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestHiveAggregationQueries.java:53]: Hmm... Should we then make the test failure advisory and non-fatal until we stabilize things? It is not a good practice to comment out tests or code in general.

**Review by mbasmanova — COMMENTED** [review]
@mshang816 Michael, looks good, but the new action seems to be failing. Please, split the changes into two comments: (1) update presto_cpp code; (2) add new actions.

**Review by highker — COMMENTED** [review]
Are we still doing anything for this PR? Looks like staled.

**Comment by mshang816** [comment]
> Are we still doing anything for this PR? Looks like staled.

This PR contains mixed changes for github actions and update on presto-native-execution. I close this one and create another PR solely for github actions

**Pull request closed without merging**